dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/datafile/ingest/IngestReport.java

```diff
@@ -141,10 +141,13 @@ public void setEndTime(Date endTime) {
 
     public String getIngestReportMessage() {
 
-        return Optional.ofNullable(errorKey)
-                .orElse(IngestError.UNKNOWN_ERROR)
-                .getErrorMessage(Optional.ofNullable(errorArguments).
-                        orElseGet(Collections::emptyList));
+        final IngestError errorKey = Optional.ofNullable(this.errorKey)
+                                             .orElse(IngestError.UNKNOWN_ERROR);
+
+        return Optional.of(errorArguments)
+                       .filter(args -> !args.isEmpty())
+                       .map(errorKey::getErrorMessage)
+                       .orElseGet(errorKey::getErrorMessage);
```

Comment:
IMO there is a problem in `IngestError.getErrorMessage(...)`. We shouldn't have to pick here, which version of `getErrorMessage` method to use. 
Maybe I am missing something, but in
```
    public String getErrorMessage(List<String> arguments) {
        return arguments.isEmpty() ? getDefaultErrorMessage() : BundleUtil.getStringFromBundle(ERROR_KEY_PREFIX + toString(), arguments.toArray());
    }
```
checking `arguments` param seems pointless and causes more confusion.
Moreover it forces that every non unknown ingest error must have some kind of arguments.